tests/src/testmaterial.cpp

```diff
@@ -228,6 +228,30 @@ auto testPlaneStrainAgainstPlaneStress(const double tol = 1e-10) {
   return t;
 }
 
+template <typename MAT>
+auto checkThrowNeoHooke(const MAT& matNH) {
+  static_assert(std::is_same_v<MAT, NeoHookeT<double>>,
+                "checkThrowNeoHooke is only implemented for Neo-Hooke material law.");
+  TestSuite t("NeoHooke Test - Checks the throw message for negative determinant of C");
+  Eigen::Vector3d E;
+  E << 2.045327969583023, 0.05875570522766141, 0.3423966429644326;
+
+  t.checkThrow<Dune::InvalidStateException>(
+      [&]() {
+        const auto moduli = (planeStress(matNH, 1e-8).template tangentModuli<StrainTags::greenLagrangian, true>(E));
```

Comment:
Maybe not repeat `planeStress(matNH, 1e-8)` but use the same object all the time?